mailer.js

```diff
@@ -278,109 +300,120 @@ module.exports = function (log) {
     if (message.redirectTo) { query.redirectTo = message.redirectTo }
     if (message.resume) { query.resume = message.resume }
 
-    var link = this.passwordResetUrl + '?' + qs.stringify(query)
+    var links = this._generateLinks(this.passwordResetUrl, message.email, query, templateName, 'reset-password')
```

Comment:
This is so much cleaner! The only other refinement I can think of here, is `content` always going to be the same for a given `templateName`? If so, would it make sense to have a `templateName` => `content` lookup table?